tests/test_validation.py

```diff
@@ -1083,3 +1054,907 @@ def test_login(self, dmge: DataModelGraphExplorer) -> None:
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchExactlyOne set rules"""
+        val_rule = "matchExactlyOne Patient.PatientID set error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            errors, _ = va_obj.cross_validation(
+                val_rule, Series(["A", "B"], index=[0, 1], name="PatientID")
+            )
+            assert len(errors) == 1
+
+            errors, _ = va_obj.cross_validation(
+                val_rule,
+                Series(["A", "B", "C", "D"], index=[0, 1, 2, 3], name="PatientID"),
+            )
+            assert len(errors) == 1
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1, "syn2": cross_val_df1},
+        ):
+            errors, _ = va_obj.cross_validation(
+                val_rule, Series(["A", "B", "C"], index=[0, 1, 2], name="PatientID")
+            )
+            assert len(errors) == 1
+
+    def test_cross_validation_match_none_set_rules_passing(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchNone set rules"""
+        val_rule = "matchNone Patient.PatientID set error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule, Series(["A", "B"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C", "D"])) == (
+                [],
+                [],
+            )
+
+    def test_cross_validation_match_none_set_rules_errors(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchNone set rules"""
+        val_rule = "matchNone Patient.PatientID set error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            errors, _ = va_obj.cross_validation(
+                val_rule,
+                Series(["A", "B", "C"], index=[0, 1, 2], name="PatientID"),
+            )
+            assert len(errors) == 1
+
+            errors, _ = va_obj.cross_validation(
+                val_rule,
+                Series(["A", "B", "C", "C"], index=[0, 1, 2, 3], name="PatientID"),
+            )
+            assert len(errors) == 1
+
+    def test_cross_validation_value_match_atleast_one_rules_passing(
```

Comment:
Per @GiaJordan 's comment about `set` v.s. `value` here: 

set vs value comes into play when you have multiple target manifests. So with value, if A,B is in target manifest 1 and C,D is in target manifest 2, and the manifest being validated has A,B,C,D, then that should not raise an error. If it was set, then both target manifests should each have A,B,C,D for the manifest to be considered valid

Can we add this test case as well? Currently this test case only covers the case where one target manifest is present. But we want to ensure that if one target manifest has A, B, and another target manifest has C, D and the tested manifest has A, B, C, D, the test can still pass

Reply:
Oh I thought I had this use case.. I'll add it in then.

Reply:
done!